importer/tasks.py

```diff
@@ -3,11 +3,55 @@
 """
 from __future__ import unicode_literals
 
+import logging
+
+from django.conf import settings
+
 from lore.celery import async
-from importer.api import import_course_from_file
+from learningresources.api import update_xanalytics
+from xanalytics import send_request, get_result, get_xanalytics_file
+
+log = logging.getLogger(__name__)
 
 
 @async.task
 def import_file(path, repo_id, user_id):
     """Asynchronously import a course."""
+    from importer.api import import_course_from_file
     import_course_from_file(path, repo_id, user_id)
+
+
+@async.task
+def populate_xanalytics_fields(course_id):
+    """
+    Initiate request to xanalytics API to get stats for a course,
+    then trigger async job to retrieve results when they become available.
+    Args:
+        course_id (int): primary key of a Course
+    """
+    if settings.XANALYTICS_URL != "":
+        token = send_request(settings.XANALYTICS_URL + "/create", course_id)
+        check_for_results.apply_async(
+            kwargs={"token": token, "wait": 5, "attempt": 1}, countdown=5)
+
+
+@async.task
+def check_for_results(token, wait, attempt):
+    """
+    Check for xanalytics results for a course.
+    Args:
+        token (string): Token received from xanalytics server.
+        wait (int): Seconds to wait before repository.
+        attempt (int): Attempt number, so we don't try forever.
+    """
+    resp = get_result(settings.XANALYTICS_URL + "/status", token)
+    if resp["status"] == "still busy" and attempt < 10:
```

Comment:
Since we are async why not wait based on  status code and potentially forever?  At the very least the retry should be an all caps variable at the top of the file so it is easy to find, adjust, and should make testing easier since you can mock it to be whatever you need.